driverapi/driverapi.go

```diff
@@ -49,3 +49,11 @@ type Driver interface {
 	// Type returns the the type of this driver, the network type this driver manages
 	Type() string
 }
+
+// JoinInfo represents a set of resources that the driver has the ability to provide during
+// join time.
+type JoinInfo struct {
+	SandboxKey      string
+	NoSandboxCreate bool
```

Comment:
IMO, Sandbox is a property of libnetwork to create and manage. I am not in favor of letting Drivers override and break that contract. If it is a Host driver specific requirement, then we should let the caller (docker in this case) to handle that situation through the provided primitives (Network and Endpoints). Hence I prefer to remove these from JoinInfo.